### PR TITLE
Add torch pointwise conv direct path

### LINKED ISSUE #18457: Reimplement torch conv ops to go direct to cudnn
Right now our conv ops in torch are a bit slow, because torch has non-standard semantics which force us to convert back and forth:

- NCHW layout
- different kernel layout
- different padding conventions

We should reimplement them in a way that goes direct to cudnn on GPU. Not only would it fix the performance regression, it's likely to end faster than vanilla PyTorch.

### PR DESCRIPTION
**Pull request by devs6186**
## Description
Closes #18457.

This adds a narrow Torch backend fast path for pointwise channels-last convolutions. For 1x1 kernels with matching input/kernel channel counts and `valid` or `same` padding, the backend now uses a direct matmul path and avoids the channels-first transpose plus Torch convolution dispatch.

The path is intentionally scoped to non-grouped pointwise channels-last convolution. Channels-first, grouped/depthwise, non-pointwise, and non-matching-channel cases continue through the existing implementation.

Checks run:
- `python -m ruff format --check keras/src/backend/torch/nn.py keras/src/ops/nn_test.py`
- `python -m ruff check keras/src/backend/torch/nn.py keras/src/ops/nn_test.py`
- `$env:KERAS_BACKEND='torch'; python -m pytest keras/src/ops/nn_test.py -k torch_channels_last_pointwise_conv_direct_path`
- `git diff --check HEAD~1..HEAD`

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by devs6186** [comment]
@gbaned This PR is currently marked as awaiting Keras engineering, and I do not see any open review feedback on my side.

The change is intentionally narrow: it only adds the Torch channels-last pointwise-conv fast path and leaves the other convolution cases on the existing path. Could you please review it when you get a chance? If anything is needed from my side, please let me know.

**Review by hertschuh — APPROVED** [review]
Thanks for the optimization!
